src/maingoblint.ml

```diff
@@ -291,7 +288,12 @@ let preprocess_files () =
 let merge_preprocessed cpp_file_names =
   (* get the AST *)
   if get_bool "dbg.verbose" then print_endline "Parsing files.";
-  let files_AST = List.map Cilfacade.getAST cpp_file_names in
+  let get_ast_and_record_deps (orig, f) =
+    let file = Cilfacade.getAST f in
+    Hashtbl.add Preprocessor.dependencies orig file.files;
```

Comment:
What's with this `orig`? CIL already provides `file.fileName`, which was fine before.

Reply:
For example:

```
orig is /home/michael/Documents/goblint-cil/analyzer/includes/stdlib.c and fileName is /home/michael/Documents/goblint-cil/analyzer/goblint_temp_78a89f/stdlib.c
orig is /home/michael/Documents/goblint-cil/analyzer/includes/pthread.c and fileName is /home/michael/Documents/goblint-cil/analyzer/goblint_temp_78a89f/pthread.c
orig is 1.c and fileName is /home/michael/Documents/goblint-cil/analyzer/goblint_temp_78a89f/1.c
```

The thing that is recorded in the merged file is also the original file name, so not using `orig` leads to an empty list of dependencies:


~~~JSON
  "files": {
    "1.c": [],
    "/usr/include/x86_64-linux-gnu/bits/byteswap.h": [],
    "/home/michael/Documents/goblint-cil/analyzer/includes/pthread.c": [],
    "/usr/include/x86_64-linux-gnu/bits/uintn-identity.h": [],
    "/home/michael/Documents/goblint-cil/analyzer/includes/stdlib.c": []
  }
~~~

vs

~~~JSON
"files": {
    "1.c": [
      "/usr/include/x86_64-linux-gnu/bits/types/struct_timeval.h",
      "/usr/include/x86_64-linux-gnu/bits/types/__sigset_t.h", "1.c",
      "/usr/include/x86_64-linux-gnu/bits/types/timer_t.h",
      "/usr/include/x86_64-linux-gnu/gnu/stubs.h",
      "/usr/include/x86_64-linux-gnu/sys/sysmacros.h",
      "/usr/include/x86_64-linux-gnu/bits/floatn-common.h",
      "/usr/include/x86_64-linux-gnu/bits/byteswap.h",
      "/usr/include/x86_64-linux-gnu/bits/typesizes.h",
      "/usr/include/endian.h", "/usr/include/x86_64-linux-gnu/sys/select.h",
      "/usr/include/x86_64-linux-gnu/bits/waitflags.h",
      "/usr/include/x86_64-linux-gnu/bits/wordsize.h",
      "/usr/include/x86_64-linux-gnu/sys/cdefs.h",
      "/usr/include/stdc-predef.h",
      "/usr/include/x86_64-linux-gnu/bits/pthreadtypes.h", "<command-line>",
      "/usr/include/x86_64-linux-gnu/bits/long-double.h",
      "/usr/include/x86_64-linux-gnu/bits/types/clockid_t.h",
      "/usr/include/x86_64-linux-gnu/bits/sysmacros.h",
      "/usr/lib/gcc/x86_64-linux-gnu/7/include/stddef.h",
      "/usr/include/x86_64-linux-gnu/bits/byteswap-16.h",
      "/usr/include/x86_64-linux-gnu/bits/uintn-identity.h",
      "/usr/include/x86_64-linux-gnu/bits/pthreadtypes-arch.h",
      "/usr/include/x86_64-linux-gnu/gnu/stubs-64.h",
      "/usr/include/x86_64-linux-gnu/bits/types/clock_t.h",
      "/usr/include/x86_64-linux-gnu/bits/waitstatus.h",
      "/usr/include/x86_64-linux-gnu/bits/endian.h",
      "/home/michael/Documents/goblint-cil/analyzer/includes/assert.h",
      "<built-in>", "/usr/include/x86_64-linux-gnu/bits/select.h",
      "/usr/include/x86_64-linux-gnu/bits/thread-shared-types.h",
      "/usr/include/x86_64-linux-gnu/sys/types.h", "/usr/include/stdlib.h",
      "/usr/include/x86_64-linux-gnu/bits/types.h",
      "/usr/include/x86_64-linux-gnu/bits/floatn.h",
      "/usr/include/x86_64-linux-gnu/bits/libc-header-start.h",
      "/usr/include/x86_64-linux-gnu/bits/types/time_t.h",
      "/usr/include/x86_64-linux-gnu/bits/types/struct_timespec.h",
      "/usr/include/x86_64-linux-gnu/bits/stdint-intn.h",
      "/usr/include/alloca.h", "/usr/include/features.h",
      "/usr/include/x86_64-linux-gnu/bits/types/sigset_t.h",
      "/usr/include/x86_64-linux-gnu/bits/stdlib-float.h"
    ],
    "/usr/include/x86_64-linux-gnu/bits/byteswap.h": [],
    "/home/michael/Documents/goblint-cil/analyzer/includes/pthread.c": [
      "/usr/include/x86_64-linux-gnu/bits/types/timer_t.h",
      "/usr/include/x86_64-linux-gnu/gnu/stubs.h", "/usr/include/sched.h",
      "/usr/include/x86_64-linux-gnu/bits/byteswap.h",
      "/usr/include/endian.h",
      "/usr/include/x86_64-linux-gnu/bits/typesizes.h",
      "/usr/include/x86_64-linux-gnu/bits/types/locale_t.h",
      "/usr/include/pthread.h",
      "/usr/include/x86_64-linux-gnu/bits/setjmp.h",
      "/usr/include/x86_64-linux-gnu/bits/wordsize.h",
      "/usr/include/x86_64-linux-gnu/sys/cdefs.h",
      "/usr/include/stdc-predef.h",
      "/usr/include/x86_64-linux-gnu/bits/cpu-set.h",
      "/usr/include/x86_64-linux-gnu/bits/pthreadtypes.h", "<command-line>",
      "/usr/include/x86_64-linux-gnu/bits/long-double.h",
      "/usr/include/x86_64-linux-gnu/bits/types/clockid_t.h",
      "/usr/include/x86_64-linux-gnu/bits/byteswap-16.h",
      "/usr/include/x86_64-linux-gnu/bits/uintn-identity.h",
      "/usr/lib/gcc/x86_64-linux-gnu/7/include/stddef.h",
      "/usr/include/x86_64-linux-gnu/bits/pthreadtypes-arch.h",
      "/usr/include/x86_64-linux-gnu/gnu/stubs-64.h",
      "/usr/include/x86_64-linux-gnu/bits/types/clock_t.h",
      "/usr/include/x86_64-linux-gnu/bits/endian.h",
      "/usr/include/x86_64-linux-gnu/bits/types/__locale_t.h", "<built-in>",
      "/usr/include/time.h",
      "/usr/include/x86_64-linux-gnu/bits/thread-shared-types.h",
      "/usr/include/x86_64-linux-gnu/bits/sched.h",
      "/home/michael/Documents/goblint-cil/analyzer/includes/pthread.c",
      "/usr/include/x86_64-linux-gnu/bits/types/struct_itimerspec.h",
      "/usr/include/x86_64-linux-gnu/bits/types.h",
      "/usr/include/x86_64-linux-gnu/bits/time.h",
      "/usr/include/x86_64-linux-gnu/bits/types/time_t.h",
      "/usr/include/x86_64-linux-gnu/bits/types/struct_timespec.h",
      "/usr/include/x86_64-linux-gnu/bits/types/struct_tm.h",
      "/usr/include/features.h"
    ],
    "/usr/include/x86_64-linux-gnu/bits/uintn-identity.h": [],
    "/home/michael/Documents/goblint-cil/analyzer/includes/stdlib.c": [
      "<command-line>", "/usr/lib/gcc/x86_64-linux-gnu/7/include/stddef.h",
      "<built-in>",
      "/home/michael/Documents/goblint-cil/analyzer/includes/stdlib.c",
      "/usr/include/stdc-predef.h"
    ]
  },
~~~